### PR TITLE
feat: development of seedlot review step 4 ownership

### LINKED ISSUE #857: Development of Seedlot Review Step 4 - Ownership
Describe the task:

Create FE of Seedlot Review, Step 4 (Ownership) and integrate endpoints as you go along.

Figma design here: https://www.figma.com/file/ROgZJQezb1DDG4AvEgjtpw/Registration-Process----Reviewer?type=design&node-id=2744-284901&mode=design&t=XxxbormmPMPHmENb-4

Image

Acceptance Criteria

- [ ]     Developed as per Figma
- [ ]     relevant documentation updated
- [ ]     tested and merged
- [ ]     Develop floater save button place holder - design to be tested by TSC staff

### PR DESCRIPTION
**Pull request by mgaseta**
# Description
Closes #857 

#### New
- Created new Ownership read and edit components
- Added them to the seedlot review view

#### Changed
- Added new  `isReview` prop on the ownership step

#### Removed
- Nothing :)

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-26-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1126-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1126-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by RMCampos** [comment]
Waiting for https://github.com/bcgov/nr-spar/pull/1177

File: frontend/src/components/SeedlotReviewSteps/Ownership/Read/index.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                {
+                  getOwnerAgencyTitle(curOwner.ownerAgency.value.description)
+                }
+              </Column>
+            </Row>
+            <Row>
+              <Column className="info-col" sm={4} md={4} lg={4}>
+                <ReadOnlyInput
+                  id={`owner-${curOwner.id}-agency`}
+                  label="Owner agency acronym"
```
Comment (RMCampos) [inline frontend/src/components/SeedlotReviewSteps/Ownership/Read/index.tsx:31]: question: Figma has "Owner agency name" here. Is this something specific for the readonly feature?

**Review by RMCampos — APPROVED** [review]
Good job! 💯 Only one question.

File: frontend/src/components/SeedlotReviewSteps/Ownership/Read/index.tsx (hunk 1 above)
Comment (mgaseta) [reply frontend/src/components/SeedlotReviewSteps/Ownership/Read/index.tsx:31]: No, there isn't, the figma print on the issue was not updated with the latest version of the design, here is the current design:
Screenshot from 2024-05-24 11-30-22

https://www.figma.com/design/ROgZJQezb1DDG4AvEgjtpw/TSC-Reviewer---Seedlot-Registration?node-id=2744-289535&t=mMMw4C32dgzpACvI-0

**Review by craigyu — APPROVED** [review]
LGTM!

**Review by craigyu — APPROVED** [review]
LGTM!

**Pull request merged**